tests/integration/cqlengine/model/test_updates.py

```diff
@@ -134,3 +134,162 @@ def test_primary_key_update_failure(self):
         m0 = TestUpdateModel.create(count=5, text='monkey')
         with self.assertRaises(ValidationError):
             m0.update(partition=uuid4())
+
+
+class ModelWithDefault(Model):
+    id          = columns.Integer(primary_key=True)
+    mf          = columns.Map(columns.Integer, columns.Integer)
+    dummy       = columns.Integer(default=42)
+
+
+class ModelWithDefaultCollection(Model):
+    id          = columns.Integer(primary_key=True)
+    mf          = columns.Map(columns.Integer, columns.Integer, default={2:2})
+    dummy       = columns.Integer(default=42)
+
+
+class ModelWithDefaultTests(BaseCassEngTestCase):
+    def setUp(self):
+        sync_table(ModelWithDefault)
+
+    def tearDown(self):
+        drop_table(ModelWithDefault)
+
+    def _assert_model_with_default_row(self, obj, expected):
+        self.assertListEqual([obj.id, obj.dummy, obj.mf], expected)
+
+    def test_value_override_with_default(self):
+        """
+        Updating a row with a new Model instance shouldn't set columns to defaults
+
+        @since 3.9
+        @jira_ticket PYTHON-657
+        @expected_result column value should not change
+
+        @test_category object_mapper
+        """
+        initial = ModelWithDefault(id=1, mf={0: 0}, dummy=0)
+        initial.save()
+
+        self._assert_model_with_default_row(ModelWithDefault.objects.all().get(),
```

Comment:
I think these calls would be clearer if they were replaced with, e.g.,

```
self.assertEqual(ModelWithDefault.objects().all().get()._as_dict(),
                 {'id': 1, 'dummy': 0, 'mf': {0:1})
```
I think that should work -- point is, there's an existing method for getting a basic data structure representing the returned model.